datanode/dehistory/snapshot/service_load_snapshot.go

```diff
@@ -104,10 +104,11 @@ func (b *Service) loadSnapshot(ctx context.Context, snapshotData compressedFileM
 		return 0, fmt.Errorf("failed to decompress and untar data: %w", err)
 	}
 
-	// TODO handle loading of data from older database version by upgrading database as snapshots
-	// TODO are loaded, for now it is expected that all snapshots are from the same database version
 	if currentDbVersion != snapshotDbVersion {
-		return 0, fmt.Errorf("snapshot database version %d does not match current database version %d", snapshotDbVersion, currentDbVersion)
+		err = b.migrateDatabaseToVersion(snapshotDbVersion)
```

Comment:
The caller calls this in a loop like this:
```
for _, history := range contiguousHistory {
	rowsCopied, err := b.loadSnapshot(ctx, history, sourceDir, dbVersion, vegaDbConn)
```
but we never update `dbVersion` to be the newer one. So if `dbVersion` is 1 so if we hit segment N with version 2 we'll call to migrate. Then we'll roll to segment N+1 which is still version 2 so we don't need to migrate, but we will because `dbVersion` is still 1.

Reply:
so we do, guess goose must be doing something sensible here to avoid multiple upgrade, will sort this

Reply:
sorted now